server/src/infra/repositories/user.repository.ts

```diff
@@ -97,8 +101,33 @@ export class UserRepository implements IUserRepository {
       stat.photos = Number(stat.photos);
       stat.videos = Number(stat.videos);
       stat.usage = Number(stat.usage);
+      stat.quotaSizeInBytes = Number(stat.quotaSizeInBytes) || null;
     }
 
     return stats;
   }
+
+  async updateUsage(id: string, { delta }: { delta: number }): Promise<void> {
+    await this.userRepository
+      .createQueryBuilder('users')
+      .update()
+      .set({ diskUsageInBytes: () => `diskUsageInBytes + ${delta}` })
+      .where({ id })
+      .execute();
+  }
+
+  async syncUsage() {
+    const subQuery = this.assetRepository
+      .createQueryBuilder('assets')
+      .select('COALESCE(SUM(exif."fileSizeInByte"), 0)')
+      .leftJoin('assets.exifInfo', 'exif')
+      .where('assets.ownerId = users.id');
```

Comment:
Does this include soft deleted assets? IMO they should be included because they still take storage space.

Reply:
It is supposed to, but I may have forgotten to add with deleted here.

Reply:
Since this is based off of exif info, maybe we should save the initial file size too when it is uploaded